src/mlpack/core/data/string_cleaning.hpp

```diff
@@ -0,0 +1,132 @@
+
+namespace mlpack {
+namespace data {
+
+class StringCleaning
+{
+ public:
+  /**
+  * Function to remove punctuation from a given vector of strings.
+  *
+  * @param input Vector of strings
+  */
+  void RemovePunctuation(std::vector<std::string>& input,
+                         const std::string& punctuation = "!\"#$%&'()*+"
+                         ",-./:;<=>?@[\\]^_`{|}~")
+  { 
+    for (auto& str : input)
+    {
+      str.erase(boost::remove_if(str, boost::is_any_of(punctuation)),
+          str.end());
+    }
+  }
+
+  /**
+  * Function to remove a set of chars from a given vector of strings.
+  * A function which accepts a function as an argument and removes the
+  * character if the funtion returns true.
+  * 
+  * @param input Vector of strings.
+  * @param fun A function which takes const char& c as input and returns bool;
+  *
+  * This can either be a function pointer or function object or a lamda
+  * function. Its return value should be of bool type.
+  *
+  * Definiation of function should be of type
+  * bool fn(const char& c)
+  */
+  template<typename ReturnType>
+  void RemoveChar(std::vector<std::string>& input,const ReturnType fun)
+  { 
+    for (auto& str : input)
+    {
+      str.erase(boost::remove_if(str, fun), str.end());
+    }
+  }
+
+  /**
+  * Function to convert given vector of strings to lower case.
+  *
+  * @param input Vector of strings
+  */
+  void LowerCase(std::vector<std::string>& input)
+  {
+    for (auto& str : input)
+    {
+      boost::algorithm::to_lower(str);
+    }
+  }
+  /**
+  * Function to convert given vector of strings to Upper case.
+  *
+  * @param input Vector of strings
+  */
+  void UpperCase(std::vector<std::string>& input)
+  {
+    for (auto& str : input)
+    {
+      boost::algorithm::to_upper(str);
+    }
+  }
+  /**
+  * Function to remove stopwords from a given vector of strings.
+  *
+  * @param input Vector of strings
+  * @param tokenizer A function that accepts a boost::string_view as
+  *                  an argument and returns a token.
+  * This can either be a function pointer or function object or a lamda
+  * function. Its return value should be a boost::string_view, a token.
+  *
+  * Definiation of function should be of type
+  * boost::string_view fn()(boost::string_view& str)
+  */
+  template<typename TokenizerType>
+  void RemoveStopWords(std::vector<std::string>& input,
+                       std::unordered_set<boost::string_view,
+                       boost::hash<boost::string_view>>stopwords,
+                       const TokenizerType tokenizer)
+  {
+    std::string copy;
+    boost::string_view token;
+    boost::string_view strView;
+    for (auto& str : input)
+    {
+      copy.clear();
+      strView = str;
+      token = tokenizer(strView);
+      while (!token.empty())
+      {
+        if (stopwords.find(token) == stopwords.end())
+        {
+          // token is not a stop word add it;
+          copy += " " + std::string(token);
```

Comment:
> The function doesn't preserve spaces. I am not sure it's correct. Did you compare this behavior with other toolkits? For example does scikit-learn preserve spaces?

Scikit-learn, doesn't have a standalone function, The have it in their pipline implementation, I compared it with NLTK library, the output of this function is
`vector<vector<string>> dataset` , After removal of stopword, they are tokenizing it,

And hence A good option would be to add a overload function in our `StringEncoding` Class which accepts `vector<vector<string>>` as input and then do the required thing, 

other one could be to introduce space and make it `vector<string> dataset`.

Reply:
Okay, in this case we are not bound to preserve delimiters.
I think we don't need the overload since it's easy to handle this in terms of tokenizers. It's enough to write a tokenizer that returns the whole string.

Reply:
> I think we don't need the overload since it's easy to handle this in terms of tokenizers. It's enough to write a tokenizer that returns the whole string.

Not sure what does this mean, You mean a function which accepts a `vector<vector<string>>` and delimiters and output a vector<string> with merger all strings ?

Reply:
No, the function still accepts `vector<string>` and a tokenizer. We needn't anything else. I mean we don't need to handle this special case.

Reply:
So i need not change anything. Right ?

`copy += " " + std::string(token);`

Reply:
Correct.